packages/wrangler/src/pages/functions/buildWorker.ts

```diff
@@ -349,17 +349,34 @@ export function buildNotifierPlugin(onEnd: () => void): Plugin {
 		name: "wrangler notifier and monitor",
 		setup(pluginBuild) {
 			pluginBuild.onEnd((result) => {
-				if (result.errors.length > 0) {
-					logger.error(
-						`${result.errors.length} error(s) and ${result.warnings.length} warning(s) when compiling Worker.`
-					);
-				} else if (result.warnings.length > 0) {
-					logger.warn(
-						`${result.warnings.length} warning(s) when compiling Worker.`
-					);
+				if (!result.errors.length && !result.warnings.length) {
+					logger.log("✨ Compiled Worker successfully");
 					onEnd();
 				} else {
-					logger.log("✨ Compiled Worker successfully");
+					if (result.errors.length > 0) {
+						const errorMessages = result.errors.reduce(
+							(acc, err) => `${acc}\n` + `▶︎ ${err.text}`,
+							""
+						);
+
+						logger.error(
+							`Got ${result.errors.length} error${result.errors.length > 1 ? "s" : ""} when compiling Worker:\n` +
+								`${errorMessages}`
+						);
+					}
+
+					if (result.warnings.length > 0) {
+						const warningMessages = result.warnings.reduce(
+							(acc, warning) => `${acc}\n` + `▶︎ ${warning.text}`,
+							""
+						);
+
+						logger.warn(
+							`Got ${result.warnings.length} warning${result.warnings.length > 1 ? "s" : ""} when compiling Worker:\n` +
+								`${warningMessages}`
+						);
+					}
```

Comment:
Could we lean on esbuild's formatting for build errors here, in the same way as we do for Workers proper? https://github.com/cloudflare/workers-sdk/blob/ab763c37454f45a6d7cd2be7963a863c49fe2d49/packages/wrangler/src/logger.ts#L149

Reply:
✅